### ISSUE TITLE
Implement CNN

### ISSUE THREAD

**Issue by DeepBlockDeepak** [DeepBlockDeepak/my-neural-network#25]
# Goal
Perform mathematical convolution by hand and allow output to compatibly feed into the `SimpleNN`'s dense layers. Start with incorporating  basic setup using parameters `in_channels`, `kernel_size`, `stride`, `padding`, `out_channels`. 

## Immediate Tasks/Ideas/deliverables:
- [x] Consider using a `Conv2D` class that specifically does the convolution logic with input params. 
- [ ] Subclass the SimpleNN for a `SimpleNNConv2D` class that calls on internal `Conv2D`
- [ ] Make simple test for ensuring operations work.

## WIP 
This first goal should involve only the initialization and forward propagation logic. Later, I need to move onto addressing back-prop and storing gradients for training.

Enhancements will look like:
- [ ] Backpropagation
- [ ] Stacking multiple convolutional Layers.
- [ ] Pooling: use max or average pooling
- [ ] Training dataset pipeline/compare to equivalent torch.nn.Conv2d
